packages/features/insights/filters/Download/Download.tsx

```diff
@@ -1,41 +1,45 @@
+import { useState } from "react";
+
 import { downloadAsCsv } from "@calcom/lib/csvUtils";
 import { useLocale } from "@calcom/lib/hooks/useLocale";
-import type { RouterOutputs } from "@calcom/trpc";
 import { trpc } from "@calcom/trpc";
-import { Dropdown, DropdownItem, DropdownMenuContent, DropdownMenuTrigger } from "@calcom/ui/components/dropdown";
 import { Button } from "@calcom/ui/components/button";
+import {
+  Dropdown,
+  DropdownItem,
+  DropdownMenuContent,
+  DropdownMenuTrigger,
+} from "@calcom/ui/components/dropdown";
 
 import { useInsightsParameters } from "../../hooks/useInsightsParameters";
 
 const Download = () => {
   const { t } = useLocale();
-  const { startDate, endDate, teamId, userId, eventTypeId, memberUserId } = useInsightsParameters();
+  const { startDate, endDate, teamId, userId, eventTypeId, memberUserId, isAll } = useInsightsParameters();
+  const [isDownloading, setIsDownloading] = useState(false);
+  const utils = trpc.useUtils();
 
-  const { data, isPending } = trpc.viewer.insights.rawData.useQuery(
-    {
-      startDate,
-      endDate,
-      teamId,
-      userId,
-      eventTypeId,
-      memberUserId,
-    },
-    {
-      refetchOnWindowFocus: false,
-      refetchOnReconnect: false,
-      retry: false,
-      staleTime: Infinity,
-      trpc: {
-        context: { skipBatch: true },
-      },
-    }
-  );
+  const handleDownloadClick = async () => {
```

Comment:
It was fetching the raw data to download as soon as the Download button was mounted!

I've fixed it by fetching the data only when the button is clicked.